File: problems/0409_longest_palindrome/0409_longest_palindrome.java
class Solution {
    public int longestPalindrome(String s) {
        int[] count = new int[128];
        for (char c : s.toCharArray()) {
            count[c]++;
        }

        int len = 0;
        for (int v : count) {
            len += (v / 2) * 2;
            if (len % 2 == 0 && v % 2 == 1) {
                len++;
            }
        }
        return len;

    }
}

File: problems/0409_longest_palindrome/0409_longest_palindrome.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        res = 0
        counts = collections.Counter(s)
        
        for count in counts.values():
            res += (count // 2) * 2
            if count % 2 == 1 and res % 2 == 0:
                res += 1
        return res